sdk/turing/router/config/router_config.py

```diff
@@ -136,7 +138,9 @@ def routes(self) -> List[Route]:
 
     @routes.setter
     def routes(self, routes: Union[List[Route], List[Dict[str, str]]]):
-        if isinstance(routes, list) and all(isinstance(route, dict) for route in routes):
```

Comment:
For the router / enricher / ensembler config, we are setting the default autoscaling policy this way:
https://github.com/caraml-dev/turing/blob/main/sdk/turing/router/config/router_config.py#L91

If the user wants to set the payload size instead, what is the expectation? Should they explicitly clear the metric and target? Or shall we clear the metric/target if payload size is set, in the `to_open_api()` function ?

Reply:
Hmm I've decided to remove the `DEFAULT_AUTOSCALING_POLICY` that exists within the SDK and to make the router / enricher / ensembler configs all require an `AutoscalingPolicy` object as a required argument during initialisation. This is to make the expectation that this field should not be nil consistent with that of the API.

> If the user wants to set the payload size instead, what is the expectation? Should they explicitly clear the metric and target? Or shall we clear the metric/target if payload size is set, in the to_open_api() function ?

As for this, there are no checks that the SDK performs to ensure that only the `metric`/`target` fields or the `payload_size` field is set. However, any configs in requests sent to the API will ultimately be validated by its own validator, which contains checks to ensure that not all the fields are set, failing which the API would return an error to the SDK during execution. This is similar to the current behaviour of the standard ensembler whereby the only checks that take place to ensure only the `route_name_path` or `experiment_mappings` field is set occur in the API.